cpp/src/barretenberg/honk/proof_system/prover.hpp

```diff
@@ -71,6 +73,9 @@ template <typename settings> class Prover {
     // Container for d + 1 Fold polynomials produced by Gemini
     std::vector<Polynomial> fold_polynomials;
 
+    Polynomial batched_quotient_Q;
```

Comment:
I wonder if a more proper way to handle this would be to have a `std::shared_ptr<Polynomial>`, rather than moving it and leaving an empty poly behind. Not a huge deal but curious about best practice here. Thoughts @adr1anh?

Reply:
Maybe an optional here could more explicitly indicate whether the entry is used?

Reply:
Shared pointer wouldn't make much sense since it's not actually shared anywhere. It also has a bit of overhead from having to do reference counting.

Reply:
A unique pointer might make sense, but the main point it in checking that the element is null or not, in which case an optional is more explicit

Reply:
@adr1anh I'm sure I'm clear on what you mean -- I haven't seen `optional` applied to a class member before, only an input/output to a function. Is this an option or am I misunderstanding you? Either way, I'll leave it as is for now since it's hardly dangerous and I'm guessing this gets refactored soon anyway :)